lib/mongoid/clients/storage_options.rb

```diff
@@ -49,7 +50,7 @@ module ClassMethods
         # @return [ Class ] The model class.
         def store_in(options)
           Validators::Storage.validate(self, options)
-          storage_options.merge!(options)
+          self.storage_options = self.storage_options.merge(options)
```

Comment:
We need to assign here so when the child is modified the parent is also not modified. If you merge!, the parent's storage_options will be modified

Reply:
:+1: